caffe2/CMakeLists.txt

```diff
@@ -999,20 +999,30 @@ elseif(USE_CUDA)
     # location, e.g.
     # `/path/to/conda/lib/python3.10/site-packages/nvidia/nvshmem`,
     # 3. Let CMake find it in the default system paths, e.g. /usr/local.
-    find_path(NVSHMEM_LIB_DIR
+    find_library(NVSHMEM_HOST_LIB
       # In pip install case, the lib suffix is `.so.3` instead of `.so`
-      NAMES libnvshmem_host.so libnvshmem_host.so.3
-      PATHS $ENV{NVSHMEM_HOME}/lib ${NVSHMEM_PY_DIR}/lib
-      DOC "The location of NVSHMEM library.")
+      NAMES nvshmem_host nvshmem_host.so.3
+      HINTS $ENV{NVSHMEM_HOME} ${NVSHMEM_PY_DIR}
+      PATH_SUFFIXES lib lib64 cuda/lib cuda/lib64 lib/x64
+      DOC "The location of NVSHMEM host library.")
+    find_library(NVSHMEM_DEVICE_LIB
+      # Device lib is a `.a` file
+      NAMES nvshmem_device
+      HINTS $ENV{NVSHMEM_HOME} ${NVSHMEM_PY_DIR}
+      PATH_SUFFIXES lib lib64 cuda/lib cuda/lib64 lib/x64
+      DOC "The location of NVSHMEM device library.")
     find_path(NVSHMEM_INCLUDE_DIR
       NAMES nvshmem.h
-      PATHS $ENV{NVSHMEM_HOME}/include ${NVSHMEM_PY_DIR}/include
+      HINTS $ENV{NVSHMEM_HOME}/include ${NVSHMEM_PY_DIR}/include
```

Comment:
We missed one replacement of PATHS -> HINTS. 
HINTS: searched before default paths.
PATHS: searched after default paths. 
Now it should be good. (I tested it locally)